UX Overhaul, add user guide

kartikcho opened the pull request:
Resolves #125 

### Changes made in this PR
- Add loading spinner when a profile is being processed (the app used to freeze the tab with no visual feedback about loading earlier).
- Add a modal explaining the profiler's different elements, events and a legend key about the event colors.
- Slightly changed landing content.

Screenshots:

![image](https://user-images.githubusercontent.com/48270786/90368106-34d80300-e087-11ea-8674-2d0a4b7e4890.png)

![image](https://user-images.githubusercontent.com/48270786/90368122-3dc8d480-e087-11ea-8236-683fac9a59ae.png)

We should update the explanations if they're not clear enough. 
Also, the legend keys look out of place in my opinion, we should think of a better layout. Maybe add another image explaining resize bar and scroll?

kartikcho committed:
Re-do landing, add guide overlay, add spinner

kartikcho committed:
Add content, css fixes, add graphics

kartikcho committed:
Fix image and legendkey responsiveness

kartikcho committed:
update profilerBrowser, add details

kartikcho commented:
What do you think about using labelled images like below?

![lanes](https://user-images.githubusercontent.com/48270786/90371197-3526cd00-e08c-11ea-932a-2330bc0f158b.png)

taneliang commented:
Cool! First impressions:

* Could we left-align the text instead? The gaps between the words make them a bit hard to read
* The text animates in 2 phases when the window is resized, like #65
* The images are too small to be seen when the window is narrow. Could we move the images above the text in those situations? 
	![image](https://user-images.githubusercontent.com/12784593/90371918-3bc03f00-e0a2-11ea-94c0-3e028e660c90.png)

> legend keys look out of place in my opinion

Yep, I agree

> What do you think about using labelled images like below?

That looks okay, but we need to explain what React measures are

kartikcho committed:
Change overlay layout, remove event legend

kartikcho commented:
> Could we left-align the text instead?

Sure

> The text animates in 2 phases

It's because of the 'lazy' transition when overlay is loaded, not really an issue.

> Could we move the images above the text in those situations?

Changed layout to single column, removed event legend key, moved images below text though.

![image](https://user-images.githubusercontent.com/48270786/90501495-139c1300-e16a-11ea-8208-b17654dfbf98.png)

kartikcho committed:
Merge branch 'master' into UX-overhaul
